src/coreclr/utilcode/clrconfig.cpp

```diff
@@ -287,77 +588,49 @@ HRESULT CLRConfig::TrimWhiteSpace(LPCWSTR wszOrig, __deref_out_z LPWSTR * pwszTr
     }
     CONTRACTL_END;
 
-    _ASSERTE(wszOrig != NULL);
-    _ASSERTE(pwszTrimmed != NULL);
-
-    // In case we return early, set [out] to NULL by default
-    *pwszTrimmed = NULL;
+    // Check if caching is disabled.
+    if (CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_DisableConfigCache) != 0)
+        return;
 
-    // Get pointers into internal string that show where to do the trimming.
-    size_t cchOrig = wcslen(wszOrig);
-    if (!FitsIn<DWORD>(cchOrig))
-        return COR_E_OVERFLOW;
-    DWORD cchAfterTrim = (DWORD) cchOrig;
-    LPCWSTR wszAfterTrim = wszOrig;
-    ::TrimWhiteSpace(&wszAfterTrim, &cchAfterTrim);
-
-    // Is input string already trimmed?  If so, save an allocation and just return.
-    if ((wszOrig == wszAfterTrim) && (cchOrig == cchAfterTrim))
-    {
-        // Yup, just return success
-        return S_OK;
-    }
-
-    if (cchAfterTrim == 0)
+#ifdef TARGET_WINDOWS
```

Comment:
Is there a good reason for this to be Windows specific?

Reply:
No. I've not gotten to a point where I am intentionally changing behavior so deferring that switch. In the next PR that adds support for `DOTNET_` I will be uncommenting this and validating it because I would expect it to "just work".